src/main/java/org/opensearch/security/ssl/http/netty/Netty4HttpRequestHeaderVerifier.java

```diff
@@ -0,0 +1,136 @@
+
+import java.util.regex.Matcher;
+
+import static com.amazon.dlic.auth.http.saml.HTTPSamlAuthenticator.API_AUTHTOKEN_SUFFIX;
+import static org.opensearch.security.filter.SecurityRestFilter.HEALTH_SUFFIX;
+import static org.opensearch.security.filter.SecurityRestFilter.PATTERN_PATH_PREFIX;
+import static org.opensearch.security.filter.SecurityRestFilter.WHO_AM_I_SUFFIX;
+import static org.opensearch.security.http.SecurityHttpServerTransport.CONTEXT_TO_RESTORE;
+import static org.opensearch.security.http.SecurityHttpServerTransport.EARLY_RESPONSE;
+import static org.opensearch.security.http.SecurityHttpServerTransport.SHOULD_DECOMPRESS;
+
+public class Netty4HttpRequestHeaderVerifier extends SimpleChannelInboundHandler<DefaultHttpRequest> {
+    private final SecurityRestFilter restFilter;
+    private final ThreadPool threadPool;
+    private final SSLConfig sslConfig;
+    private final boolean injectUserEnabled;
+    private final boolean passthrough;
+
+    public Netty4HttpRequestHeaderVerifier(
+        SecurityRestFilter restFilter,
+        NamedXContentRegistry xContentRegistry,
+        ThreadPool threadPool,
+        HttpHandlingSettings handlingSettings,
+        Settings settings
+    ) {
+        this.restFilter = restFilter;
+        this.threadPool = threadPool;
+
+        this.injectUserEnabled = settings.getAsBoolean(ConfigConstants.SECURITY_UNSUPPORTED_INJECT_USER_ENABLED, false);
+        boolean disabled = settings.getAsBoolean(ConfigConstants.SECURITY_DISABLED, false);
+        if (disabled) {
+            sslConfig = new SSLConfig(false, false);
+        } else {
+            sslConfig = new SSLConfig(settings);
+        }
+        boolean client = !"node".equals(settings.get(OpenSearchSecuritySSLPlugin.CLIENT_TYPE));
+        this.passthrough = client || disabled || sslConfig.isSslOnlyMode();
+    }
+
+    @Override
+    public void channelRead0(ChannelHandlerContext ctx, DefaultHttpRequest msg) throws Exception {
+        // DefaultHttpRequest should always be first and contain headers
+        ReferenceCountUtil.retain(msg);
+
+        if (passthrough) {
+            ctx.fireChannelRead(msg);
+            return;
+        }
+
+        // Start by setting this value to false, only requests that meet all the criteria will be decompressed
+        ctx.channel().attr(SHOULD_DECOMPRESS).set(Boolean.FALSE);
+
+
+        final Netty4HttpChannel httpChannel = ctx.channel().attr(Netty4HttpServerTransport.HTTP_CHANNEL_KEY).get();
+        Matcher matcher = PATTERN_PATH_PREFIX.matcher(msg.uri());
+        final String suffix = matcher.matches() ? matcher.group(2) : null;
+        if (API_AUTHTOKEN_SUFFIX.equals(suffix)) {
+            // TODO:           I think this is going to create problems - we should have a sensible size limit, not prevention of
+            // TODO_CONTINUED: decompression - it will prevent valid response bodies that are gzip'ed from being usable no?
+            ctx.fireChannelRead(msg);
```

Comment:
Calling this comment out - when there is a response body that we want to read, we need to decompress it.  For the ACS endpoint we shouldn't reject requests because they are compressed, no?